common/lib/api-client/json-api.test.js

```diff
@@ -40,4 +46,92 @@ describe('Back-end API client', function() {
       expect(apiMock.isDone()).to.be.true
     })
   })
+
+  describe('Models', function() {
+    const testCases = {
+      move: [
+        {
+          method: 'find',
+          httpMock: 'get',
+          mockPath: '/1',
+          args: '1',
+        },
+        {
+          method: 'findAll',
+          httpMock: 'get',
+        },
+      ],
+      person: [
+        {
+          method: 'create',
+          httpMock: 'post',
+          args: {},
+        },
+      ],
+      gender: [
+        {
+          method: 'findAll',
+          httpMock: 'get',
+        },
+      ],
+      ethnicity: [
+        {
+          method: 'findAll',
+          httpMock: 'get',
+        },
+      ],
+      assessment_question: [
+        {
+          method: 'findAll',
+          httpMock: 'get',
+        },
+      ],
+      location: [
+        {
+          method: 'findAll',
+          httpMock: 'get',
+        },
+        {
+          method: 'find',
+          httpMock: 'get',
+          mockPath: '/1',
+          args: '1',
+        },
+      ],
+    }
+
+    forEach(models, (model, modelName) => {
+      describe(`${modelName} model`, function() {
+        forEach(testCases[modelName], testCase => {
+          describe(`#${testCase.method}()`, function() {
+            beforeEach(function() {
+              const apiPath =
+                get(model, 'options.collectionPath') || pluralize(modelName)
+              const fixture = require(`${fixturesPath}/${modelName}.${testCase.method}.json`)
+
+              nock(apiUrl)
+                .intercept(
+                  `/${apiPath}${testCase.mockPath || ''}`,
+                  testCase.httpMock
+                )
+                .reply(200, fixture)
+            })
+
+            it('should contain correct attributes', async function() {
+              const { data } = await jsonApi[testCase.method](
+                modelName,
+                testCase.args
+              )
+
+              if (Array.isArray(data)) {
+                data.forEach(item => checkProperties(item, model.attributes))
```

Comment:
could just do 
```
flatten([data]).forEach(item => checkProperties(item, model.attributes))
```

But maybe thats a little magic!

Reply:
Ooo, think I prefer that, reads shorted. Could use spread too...